src/main/scala/com/lightbend/kafka/scala/streams/typesafe/derivations/package.scala

```diff
@@ -0,0 +1,85 @@
+package com.lightbend.kafka.scala.streams.typesafe
+
+import java.nio.ByteBuffer
+
+import org.apache.kafka.common.serialization._
+import org.apache.kafka.streams.Consumed
+import org.apache.kafka.streams.kstream.{Joined, Materialized, Produced,
+  Serialized}
+import org.apache.kafka.streams.processor.StateStore
+
+/** Many default derivations, provided as standalone universal traits that
+  * can be mixed in anywhere.
+  *
+  * Also provides a [[derivations.Default]] object that can be used directly
+  * extending all these mix-in traits.
+  *
+  * @author Santiago Saavedra (ssaavedra@openshine.com)
+  */
+package object derivations {
+
+  trait materialized extends Any {
+    implicit def materializedFromSerdes[K, V, S <: StateStore]
+    (implicit key: Serde[K], value: Serde[V]): Materialized[K, V, S] = {
+      util.Materialized[K, V]
+    }
+  }
+
+  trait serialized extends Any {
+    implicit def serializedFromSerdes[K, V](implicit key: Serde[K],
+                                            value: Serde[V])
+    : Serialized[K, V] = {
+      Serialized.`with`(key, value)
+    }
+  }
+
+  trait produced extends Any {
+    implicit def producedFromSerdes[K, V](implicit key: Serde[K],
+                                          value: Serde[V])
+    : Produced[K, V] = {
+      Produced.`with`(key, value)
+    }
+  }
+
+  trait consumed extends Any {
+    implicit def consumedFromSerdes[K, V](implicit key: Serde[K],
+                                          value: Serde[V])
+    : Consumed[K, V] = {
+      Consumed.`with`(key, value)
+    }
+  }
+
+  trait joined extends Any {
+    implicit def joinedFromSerdes[K, V, VO](implicit key: Serde[K],
+                                            value: Serde[V], value2: Serde[VO])
+    : Joined[K, V, VO] = {
+      Joined.`with`(key, value, value2)
+    }
+  }
```

Comment:
Does this allow us to get a `Joined` by specifying 2 of the serdes while the third one is picked from config ? We can do that with `unsafely` by explicitly specifying the `Joined` parameter. But can we do this through this implicit ?

Reply:
I am not sure I understand this question.

This implicit definition will generate a Joined[K, V, VO] instance that holds the three appropriate Serde if there is one for each K, V, VO type. More convolutely said, this `implicit def` provides a proof for instantiation of a Joined[K, V, VO] instance upon witnessing an instance of each `Serde[K]`, `Serde[V]` and `Serde[VO]`.

If you don't provide a witness for some of the Serde values, this definition will not be considered.

But you can always create your own Joined instance using the Joined constructor, this definition just provides a *sane* way for automatically generating such an instance. But if you wish otherwise, you can always provide your own Joined, even if you can implicitly have another by providing it explicitly to the underlying call as I demonstrate on https://github.com/lightbend/kafka-streams-scala/pull/52/files/19400f78f4ae80adae946e266ea6d19c96b150f4#diff-ed482980f14897992a99a62b8bfe9310R119

You can always provide explicitly an implicit value at a function call site. But this `implicit def` only generates Joined instances that are type checked against all their parameters.

If you don't want that to happen and you wish lesser type constraints you can avoid importing it into your scope (as you see, the modularity in this file allows you to construct your DIY-trait with the implicits you may wish to provide while forcing others to be used at call time or provided as implicit vals on the code).